### PR TITLE
fix(agw): s1aptest TestAttachDetachRarTcpDataWithHE is flaky / or broken

### LINKED ISSUE #6254: [s1aptest] TestAttachDetachRarTcpDataWithHE is flaky / or broken
### Your Environment

- **Version:** master

### Describe the Issue
TestAttachDetachRarTcpDataWithHE seems to be flaky.
See here for the test insights: https://app.circleci.com/insights/github/magma/magma/workflows/agw/tests?reporting-window=last-90-days&branch=master

* 0% success rate
* failed in the last 7 runs out of 10
Screen Shot 2021-04-20 at 4 43 36 PM

### PR DESCRIPTION
**Pull request by vikramgreddy**
Signed-off-by: Vikram G <vikram.gurrappagaru@radisys.com>

## Summary
Fixes #6254
Default flows in table 0 to 4

## Test Plan
make integ_test

### CODE REVIEW

File: lte/gateway/python/integ_tests/defs.mk [hunk 1]
```diff
@@ -125,7 +125,8 @@ s1aptests/test_attach_ul_udp_data_with_multiple_service_restart.py \
 s1aptests/test_attach_ul_udp_data_with_pipelined_restart.py \
 s1aptests/test_attach_ul_udp_data_with_sessiond_restart.py \
 s1aptests/test_service_req_ul_udp_data_with_mme_restart.py \
-s1aptests/test_attach_detach_setsessionrules_tcp_data.py
+s1aptests/test_attach_detach_setsessionrules_tcp_data.py \
+s1aptests/test_attach_detach_rar_tcp_he.py
```
Comment (VinashakAnkitAman) [inline lte/gateway/python/integ_tests/defs.mk]: Please remove the same test case name commented in line 265
-# s1aptests/test_attach_detach_rar_tcp_he.py \ GitHubIssue 6254

File: lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py [hunk 2]
```diff
@@ -613,8 +613,8 @@ def verify_flow_rules_deletion(self):
             dpath, {"table_id": self.SPGW_TABLE},
         )
         assert(
-            len(flows) == 2
-        ), "There should only be 2 default table 0 flows"
+            len(flows) == 4
+        ), "There should only be 4 default table 0 flows"
```
Comment (VinashakAnkitAman) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py]: Is it needed for all test cases? Have we recently introduced 2 new default flow rules which would be there even after UE is detached?

**Comment by VinashakAnkitAman** [comment]
**The PR #11101 causing this issue of increased default flow rule count has been reverted as part of the PR #12390**
Please update this PR once the PR #12390 is merged, to include the test case in sanity after re-verification.

File: lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py (hunk 2 above)
Comment (VinashakAnkitAman) [reply lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py]: This was after effect of PR #11101

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
@VinashakAnkitAman @vikramgreddy what is the plan with this PR? looks like it has been open for a while?

**Comment by VinashakAnkitAman** [comment]
> What is supposed to have fixed this test? I just tried running it on master and it fails for me:
> 
> ```
> ...
> ======================================================================
> FAIL: Attach/detach + send ReAuth Req to session manager with a
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/vagrant/magma/lte/gateway/python/integ_tests/s1aptests/test_attach_detach_rar_tcp_he.py", line 222, in test_attach_detach_rar_tcp_data_with_he
>     assert len(uplink_flows) > 1, "Uplink flow missing for UE"
> AssertionError: Uplink flow missing for UE
> -------------------- >> begin captured logging << --------------------
> root: ERROR: Could not load magmad yml config for mconfig modules
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "GET /stats/switches HTTP/1.1" 200 34
> root: INFO: Adding subscriber : IMSI001010000000001
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "GET /stats/switches HTTP/1.1" 200 34
> root: DEBUG: s1ap response expected, received: 12, 12
> root: DEBUG: s1ap message expected, received: 68, 68
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "POST /stats/flow/121132577225037 HTTP/1.1" 200 23
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "POST /stats/flow/121132577225037 HTTP/1.1" 200 23
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "POST /stats/flow/121132577225037 HTTP/1.1" 200 23
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "POST /stats/flow/121132577225037 HTTP/1.1" 200 23
> urllib3.connectionpool: DEBUG: Starting new HTTP connection (1): 192.168.60.142:3333
> urllib3.connectionpool: DEBUG: http://192.168.60.142:3333 "POST /stats/flow/121132577225037 HTTP/1.1" 200 23
> root: INFO: Deleting subscriber : IMSI001010000000001
> --------------------- >> end captured logging << ---------------------
> 
> ----------------------------------------------------------------------
> Ran 1 test in 90.235s
> 
> FAILED (failures=1)
> Makefile:49: recipe for target 'integ_test' failed
> make: *** [integ_test] Error 1
> ```
> 
> Note that `test_attach_detach.py` works for me, so my set-up works at least partially:
> 
> ```
> ...
> *************NbUiNbuHdlUeRadioCapMsg: Send UE cap msg SUCCESS*********
> PDN_TYPE: 1
> IPv4 PDN_ADDRESS:192 168 128 13
> ************************* Running UE detach for UE id  2
> Deleting Ue Context from Traffic Handler
> ************************* send SCTP SHUTDOWN
> Keys left in Redis (list should be empty)[
>   
> ]
> Entries in s1ap_imsi_map (should be zero): 0
> Entries left in hashtables (should be zero): 0
> Entries in mme_ueip_imsi_map (should be zero): 0
> .
> ----------------------------------------------------------------------
> Ran 1 test in 8.342s
> 
> OK
> sleep 1
> ```

@voisey For me the test case is running fine on latest master. I ran it 3-4 times in loop. Not sure if  it got fixed meanwhile or the issue is flaky. 
```
(python) vagrant@magma-test:~/magma/lte/gateway/python/integ_tests$ make integ_test TESTS=s1aptests/test_attach_detach_rar_tcp_he.py
. /home/vagrant/build/python/bin/activate
echo "Running test: s1aptests/test_attach_detach_rar_tcp_he.py"
Running test: s1aptests/test_attach_detach_rar_tcp_he.py
timeout --foreground -k 930s 900s sudo -E PATH=/home/vagrant/build/python/bin:/usr/lib/ccache:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/go/bin PYTHONPATH=:/home/vagrant/s1ap-tester/bin:/home/vagrant/s1ap-tester/bin /home/vagrant/build/python/bin/nosetests  --with-xunit --xunit-file=/var/tmp/test_results/test_attach_detach_rar_tcp_he.xml -x -s s1aptests/test_attach_detach_rar_tcp_he.py || (echo "fail" > /home/vagrant/magma/test_status.txt && exit 1)
Test Case Execution Count: 1
Start time 15:02:15
tag: AUTH_TYPE tagVal: MILENAGE
************************* Enb tester config

**********************TPT open server******** 0

 Server Socket FD =[13]
************* Testing *Time elapsed(in usec): for timer expiry               :8770 usec
AGW is stateless
APN Correction configured
Health service is disabled
Using subscriber IMSI IMSI001010000000001
Using IMEI 3805468432113171
************************* UE device config for ue_id  1
************************* Configuring IP block
************************* Waiting for IP changes to propagate
************************* S1 setup
************************* UE App config
********************** Running End to End attach for  UE id  1
ue id 1 found

*************NbUiNbuHdlUeRadioCapMsg: Send UE cap msg SUCCESS*********
PDN_TYPE: 1
IPv4 PDN_ADDRESS:192 168 128 12
Sleeping for 5 seconds
********************** Sending RAR for  IMSI001010000000001
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 22459    0 22459    0     0  5483k      0 --:--:-- --:--:-- --:--:-- 5483k
Sending Policy RAR message to session manager

Sleeping for 5 seconds
************** Sending activate dedicated EPS bearer context accept

Checking for uplink flow
Get uplink flows: attempt  0
Get uplink flows: attempt  1
Checking for downlink flow
Get downlink flows: attempt  0
Sleeping for 5 seconds
Connecting to host 192.168.129.42, port 5001
[ 31] local 192.168.128.12 port 52871 connected to 192.168.129.42 port 5001
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[ 31]   0.00-1.00   sec  18.6 KBytes   152 Kbits/sec    0   5.62 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[ 31]   0.00-1.00   sec  18.6 KBytes   152 Kbits/sec    0             sender
[ 31]   0.00-1.00   sec  16.9 KBytes   138 Kbits/sec                  receiver

iperf Done.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 27012    0 27012    0     0  3768k      0 --:--:-- --:--:-- --:--:-- 4396k
********************** Deleting dedicated bearer for IMSI 001010000000001
Sending DeleteBearer request to spgw service
******************* Received deactivate eps bearer context
************* Sending deactivate EPS bearer context accept

Sleeping for 5 seconds
********************** Running UE detach for UE id  1
Deleting Ue Context from Traffic Handler
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 27568    0 27568    0     0  5384k      0 --:--:-- --:--:-- --:--:-- 6730k
Checking if all uplink/downlink flows were deleted
************************* send SCTP SHUTDOWN
Keys left in Redis (list should be empty)[

]
Entries in s1ap_imsi_map (should be zero): 0
Entries left in hashtables (should be zero): 0
Entries in mme_ueip_imsi_map (should be zero): 0
.
----------------------------------------------------------------------
Ran 1 test in 57.002s

OK

```

@vikramgreddy Please fix merge conflict and reverify the change once more

**Comment by VinashakAnkitAman** [comment]
@vikramgreddy Closing this PR as the changes were already merged to master as part of another PR #12581

**Pull request closed without merging**